src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -171,54 +181,52 @@ class LdapAccessPolicyDAO(
           .asJava))
       .toSeq
 
-    for {
-      stream <- executeLdap(
-        IO(ldapSearchStream(resourceTypeDn(resourceTypeName), SearchScope.ONE, filters: _*)(et => unmarshalResourceAuthDomain(et, resourceTypeName))))
-      res <- IO.fromEither(stream.parSequence.leftMap(s => new WorkbenchException(s)))
-    } yield res.toSet
+    ldapSearchStream(resourceTypeDn(resourceTypeName), SearchScope.ONE, filters: _*)(et => unmarshalResourceAuthDomain(et, resourceTypeName))
   }
 
-  private def unmarshalResourceAuthDomain(entry: Entry, resourceTypeName: ResourceTypeName): Either[String, Resource] = {
+  private def unmarshalResourceAuthDomain(entry: Entry, resourceTypeName: ResourceTypeName): IO[Resource] = {
     val authDomains = getAttributes(entry, Attr.authDomain).map(WorkbenchGroupName)
-    for {
-      resourceId <- getAttribute(entry, Attr.resourceId).toRight(s"${entry.getDN} attribute missing: ${Attr.resourceId}. ")
-    } yield Resource(resourceTypeName, ResourceId(resourceId), authDomains)
+    val resourceIdOption = getAttribute(entry, Attr.resourceId)
+    resourceIdOption match {
+      case None => IO.raiseError(new WorkbenchException(s"${entry.getDN} attribute missing: ${Attr.resourceId}."))
+      case Some(resourceId) => IO.pure(Resource(resourceTypeName, ResourceId(resourceId), authDomains))
+    }
```

Comment:
You don't need to pattern match. You can `IO.fromEither(yourEither).leftMap(s => new WorkbenchException(???))`